### ISSUE TITLE
Concatenation issue when setting OSPF redistribute process id in ios_bgp_global

### ISSUE THREAD

**Issue by danialre** [ansible-collections/cisco.ios#780]
##### SUMMARY
When using the ios_bgp_global module, setting the OSPF process ID in the redistribute section results in a module error "an only concatenate str (not \"int\") to str".

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ios_bgp_global

##### ANSIBLE VERSION
```
ansible [core 2.14.3]
  config file = None
  configured module search path = ['/Users/me/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/me/Library/Python/3.9/lib/python/site-packages/ansible
  ansible collection location = /Users/me/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/me/Library/Python/3.9/bin/ansible
  python version = 3.9.6 (default, Oct 18 2022, 12:41:40) [Clang 14.0.0 (clang-1400.0.29.202)] (/Library/Developer/CommandLineTools/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION
```
# /Users/me/.ansible/collections/ansible_collections
Collection Version
---------- -------
cisco.ios  4.4.0
```

##### CONFIGURATION
```
CONFIG_FILE() = None
```

##### OS / ENVIRONMENT
Tested on macOS Ventura 13.2, as well as Ubuntu 22.04 with Python 3.10 and ansible core 2.14.1

##### STEPS TO REPRODUCE
```
# simple playbook
---
- hosts: all
  gather_facts: no
  connection: network_cli
  tasks:
  - name: Render BGP config
    cisco.ios.ios_bgp_global:
      state: rendered
      config:
        as_number: 64512
        redistribute:
        - ospf:
            process_id: 64512
```

```
# inventory.yml for above playbook
---
all:
  hosts:
    test:
      ansible_host: 10.10.10.10
      ansible_network_os: cisco.ios.ios
```

##### EXPECTED RESULTS
A successful play with an IOS config that looks something like this:
```
router bgp 64512
 address-family ipv4
  redistribute ospf 64512
```

##### ACTUAL RESULTS
A failed play on the example playbook. Extra verbosity doesn't seem to give any details on the error message.
```
PLAY [all] ********************************************************************************************************************************************************************************************

TASK [Render BGP config] ******************************************************************************************************************************************************************************
fatal: [test]: FAILED! => {"changed": false, "module_stderr": "can only concatenate str (not \"int\") to str", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}

PLAY RECAP ********************************************************************************************************************************************************************************************
test                       : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

**Comment by bentole** [comment]
Hi @danialre! Thanks for reporting this. I have somewhat selfish reasons to follow up on defects in the bgp modules 😆 Had a peek and it's easy to resolve, but there are some other glitches too that needs to be fixed for the redistribution to be a success. I'll have a PR up in a jiff. @KB-perByte @NilashishC, just assign this one to me if it's ok with you guys.

**Comment by bentole** [comment]
Hi @danialre ! Just raised a PR fixing some issues, but i noticed you're using `cisco.ios.ios_bgp_global`. Consider using `cisco.ios.ios_bgp_address_family` instead since you're inside an AF. I believe it's wise to do this inside an AF, as you do 😃

**Comment by danialre** [comment]
I was using `cisco.ios.ios_bgp_global` since I was also setting Router ID, neighbors, timers, etc.
I also tried `ios.ios_bgp_address_family` but it did not generate the IOS redistribution config.

Sample playbook:
```
---
- hosts: all
  gather_facts: no
  connection: network_cli
  tasks:
  - name: Render BGP config
    cisco.ios.ios_bgp_address_family:
      state: rendered
      config:
        as_number: 64512
        address_family:
        - afi: ipv4
          redistribute:
          - ospf:
              process_id: 64512
              metric: 50
    register: rbgp

  - name: Debug
    debug:
      msg: "{{ rbgp.rendered }}"
```

Output:
```
PLAY [all] ***************************************************************************************************

TASK [Render BGP config] *************************************************************************************
ok: [test]

TASK [Debug] *************************************************************************************************
ok: [test] => {
    "msg": []
}

PLAY RECAP ***************************************************************************************************
test                       : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

**Comment by bentole** [comment]
@danialre Yep, the same defects exist in `ios_address_family` as well. You're hitting `no commands generated`. Should be fixed by PR #783. Take care!

**Comment by KB-perByte** [comment]
Closing this issue as https://github.com/ansible-collections/cisco.ios/pull/783 is merged
